src/IdentityServer/Validation/Default/AuthorizeRequestValidator.cs

```diff
@@ -874,27 +874,24 @@ private async Task<AuthorizeRequestValidationResult> ValidateOptionalParametersA
         }
 
         //////////////////////////////////////////////////////////
-        // check session cookie
+        // session id
         //////////////////////////////////////////////////////////
-        if (_options.Endpoints.EnableCheckSessionEndpoint)
+        if (request.Subject.IsAuthenticated())
         {
-            if (request.Subject.IsAuthenticated())
+            var sessionId = await _userSession.GetSessionIdAsync();
+            if (sessionId.IsPresent())
             {
-                var sessionId = await _userSession.GetSessionIdAsync();
-                if (sessionId.IsPresent())
-                {
-                    request.SessionId = sessionId;
-                }
-                else
-                {
-                    LogError("Check session endpoint enabled, but SessionId is missing", request);
-                }
+                request.SessionId = sessionId;
             }
             else
             {
-                request.SessionId = ""; // empty string for anonymous users
+                LogError("SessionId is missing", request);
```

Comment:
I don't think it is normally possible to not have a session id, so I think logging this as an error and then continuing is reasonable.